packages/device-client/src/lib/device.store.ts

```diff
@@ -43,16 +53,20 @@ export const deviceClient = (config: ConfigOptions) => {
        * @async
        * @function get
        * @param {RetrieveOathQuery} query - The query used to retrieve Oath devices.
-       * @returns {Promise<OAthResponse>} - A promise that resolves to the retrieved data or undefined if the response is not valid.
+       * @returns {Promise<OAthResponse | { error: unknown }>} - A promise that resolves to the retrieved data or an error object if the response is not valid.
        */
-      get: async function (query: RetrieveOathQuery) {
-        const response = await store.dispatch(endpoints.getOAthDevices.initiate(query));
+      get: async function (query: RetrieveOathQuery): Promise<OathDevice[] | { error: unknown }> {
+        try {
+          const response = await store.dispatch(endpoints.getOAthDevices.initiate(query));
 
-        if (!response || !response.data) {
-          return undefined;
-        }
+          if (!response || !response.data || !response.data.result) {
+            throw new Error('response did not contain data');
+          }
 
-        return response.data;
+          return response.data.result;
+        } catch (error) {
+          return { error };
```

Comment:
@ryanbas21 Should this throw the `GenericError` type that we settled on in DaVinci Client? My thought is that we should probably be consistent with errors throughout this SDK suite.